app/src/main/java/com/example/una/fragments/HomePageFragment.java

```diff
@@ -137,10 +137,18 @@ public void onSuccess(int statusCode, Header[] headers, JSONObject response) {
                     JSONArray groupsArr = response.getJSONArray("groups");
                     JSONObject groupsObj = groupsArr.getJSONObject(0);
                     JSONArray organizations = groupsObj.getJSONArray("organizations");
+
+                    String prevName = "";
                     for (int i = 0; i < organizations.length(); i++) {
                         JSONObject rankedCharity = organizations.getJSONObject(i);
-                        Charity charity = new Charity(rankedCharity.getJSONObject("organization"));
-                        featured.add(charity);
+                        Charity charity = fromJSON(rankedCharity.getJSONObject("organization"));
+
+                        // make sure there are no duplicates in the data
+                        String newName = charity.name;
```

Comment:
are the duplicates always adjacent?

Reply:
Its possible that they are not, but in the responses that I looked at, they were always adjacent

Reply:
this will work for now but might need to be changed in future if we find that duplicates are not always adjacent.